diff --git a/rel/files/riak b/rel/files/riak
index 3ffd6e1..2ca206d 100755
--- a/rel/files/riak
+++ b/rel/files/riak
@@ -37,7 +37,7 @@ if ([ "$RUNNER_USER" ] && [ "x$WHOAMI" != "x$RUNNER_USER" ]); then
         exit 1
     fi
     echo "Attempting to restart script through sudo -H -u $RUNNER_USER"
-    exec sudo -H -u $RUNNER_USER -i $RUNNER_SCRIPT_DIR/$RUNNER_SCRIPT $@
+    exec sudo -H -u $RUNNER_USER $RUNNER_SCRIPT_DIR/$RUNNER_SCRIPT $@
 fi
 
 # Warn the user if ulimit -n is less than 1024
@@ -212,7 +212,7 @@ case "$1" in
     attach)
         if [ "$2" = "-f" ]; then
           echo "Forcing connection..."
-        else  
+        else
           # Make sure a node is running
           RES=`ping_node`
           ES=$?
diff --git a/rel/files/riak-admin b/rel/files/riak-admin
index 4711bde..fee8b61 100755
--- a/rel/files/riak-admin
+++ b/rel/files/riak-admin
@@ -30,7 +30,7 @@ if ([ "$RUNNER_USER" ] && [ "x$WHOAMI" != "x$RUNNER_USER" ]); then
         exit 1
     fi
     echo "Attempting to restart script through sudo -H -u $RUNNER_USER"
-    exec sudo -H -u $RUNNER_USER -i $RUNNER_SCRIPT_DIR/$RUNNER_SCRIPT $@
+    exec sudo -H -u $RUNNER_USER $RUNNER_SCRIPT_DIR/$RUNNER_SCRIPT $@
 fi
 
 # Make sure CWD is set to runner base dir
diff --git a/rel/files/search-cmd b/rel/files/search-cmd
index d80fdde..167e246 100755
--- a/rel/files/search-cmd
+++ b/rel/files/search-cmd
@@ -25,7 +25,7 @@ if [ "$RUNNER_USER" -a "x$LOGNAME" != "x$RUNNER_USER" ]; then
         exit 1
     fi
     echo "Attempting to restart script through sudo -H -u $RUNNER_USER"
-    exec sudo -H -u $RUNNER_USER -i $RUNNER_SCRIPT_DIR/$RUNNER_SCRIPT $@
+    exec sudo -H -u $RUNNER_USER $RUNNER_SCRIPT_DIR/$RUNNER_SCRIPT $@
 fi
 
 # Make sure CWD is set to runner base dir
