--- c_src.orig/js/src/config/SunOS5.10.mk	2008-01-19 18:12:59.000000000 -0800
+++ c_src/js/src/config/SunOS5.10.mk	2010-02-11 07:44:41.000000000 -0800
@@ -46,5 +46,15 @@
 INCLUDES += -I/usr/sfw/include/mozilla/nspr
 OTHER_LIBS += -L/usr/sfw/lib/mozilla -R/usr/sfw/lib/mozilla
 
-CC=gcc
+CC         = gcc
+LD         = gcc
+MKSHLIB    = $(LD) -shared
+OS_CFLAGS += -fPIC
+CFLAGS    += -fPIC
+ifneq (86,$(findstring 86,$(OS_RELEASE)))
+ASFLAGS   += -xarch=v9 -DULTRA_SPARC
+DEFINES   += -DULTRA_SPARC
+OS_CFLAGS += -Wa,-xarch=v9,-DULTRA_SPARC
+CFLAGS    += -DULTRA_SPARC
+endif
 
